compiler/damlc/tests/daml-test-files/InterfaceGuardedNotExtended.daml

```diff
@@ -0,0 +1,127 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF-FEATURE DAML_INTERFACE
+-- @UNTIL-LF-FEATURE DAML_INTERFACE_EXTENDED
+-- @ERROR Requires in Daml interfaces are only available with --target=1.dev
```

Comment:
Hm I'd like to see what happens for `exerciseGuarded`, but we get this error because `interface SubToken requires Token`. Could you remove that interface and see what error we get?

Reply:
When I remove that, the error is
```
unknown value: a85bc8de7d4fd3acb3bb196c20ba2600f4b2bd4be4a3cb91d2ae3f467463383d:DA.Internal.Template.Functions:exerciseGuarded
```

Like we discussed earlier - this is caused by the guard pruning code we wrote in #14770 , which causes these hard to read errors when you try to use guards in a version which doesn't support them

Reply:
Thanks for reminding me. I wonder if we could throw an error in `LFConversion.convertExpr` when we see a _call_ of   `exerciseGuarded`, but it's not remotely a blocker.